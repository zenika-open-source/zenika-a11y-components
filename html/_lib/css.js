/**
 * As much as possible we will use DOM native attributes to reflect the state of
 * our elements and CSS will use the attribute selector `[â€¦]` to handle the
 * appropriate look and feel. However some special cases are easier to handle
 * through a simple class name:
 *
 *  - When no native attribute exist to express the element state
 *  - When no native attribute allows to express element semantic
 *
 * So in order to ease customization, those classes name will live in that centralized module
 *
 * @typedef {Object<string, string>} CSSClasses
 */
export default {
  /**
   * Necessary to express a "soft" hide (when an element is visually
   * hidden but still perceivable by screen reader). This is especially
   * useful when transitioning from a "hard" hide (`display:none`) to
   * a fully visible element
   *
   * Basically, to be fully accessible, a show/hide full sequence has to be:
   *
   * 1. The element isn't in the DOM
   * 2. The element is in the DOM, but is hidden "hard" (`display:none` or attribute `hidden`)
   * 3. The element is in the DOM, but is hidden "soft" (not visible, but readable by screenreaders)
   * 4. The element is in the DOM and visible
   *
   * If you want to apply a CSS transition it can only be done between step 3 and 4
   * so we need a dedicated CSS class to materialized step 3.
   *
   * @property {string}
   */
  HIDE: 'hidden',

  /**
   * An overlay element is something put on top of other content. There is no
   * normalized combination of attribute to identify such element so we use that
   * class name as a generic way to identify overlays.
   *
   * @property {string}
   */
  OVERLAY: 'overlay',

  /**
   * A dialogue box (`[role='dialog']`) is always put inside an overlay
   * that must be identified as such with this class name.
   *
   * @property {string}
   */
  DIALOG_OVERLAY: 'dialog'
}
