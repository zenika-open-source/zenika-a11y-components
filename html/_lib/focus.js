// Focus trap state -----------------------------------------------------------
let FOCUS_TRAP = null
let FOCUS_DIRECTION = 'first'

// Get focus direction out of keyboard events ---------------------------------
document.addEventListener('keydown', (e) => e.key === 'Tab'
  && (FOCUS_DIRECTION = (e.shiftKey && 'last') || 'first'))
document.addEventListener('keyup', (e) => e.key === 'Tab'
  && (FOCUS_DIRECTION = 'first'))

// Utils ----------------------------------------------------------------------
/** Try to focus a child on a given node.
 *
 * @param {HTMLElement} node
 *   The parent where to focus a child
 * @param {`first`|`last`} firstOrLast
 *   The child to focus, either the first or the last focusable one
 * @returns {boolean}
 */
function focusChild(node, firstOrLast) {
  const children = Array.from(node.children)

  if (firstOrLast === 'last') {
    children.reverse()
  }

  return children.some((child) => (
    child.focus()
    || document.activeElement === child
    || focusChild(child, firstOrLast)
  ))
}

// Main API -------------------------------------------------------------------
/** Release the focus from the current trap
 * @param {boolean} restoreOriginalFocus
 */
export function release(restoreOriginalFocus) {
  FOCUS_TRAP?.release(restoreOriginalFocus)
}

/** Trap the focus inside a given node
 *
 * @param {HTMLElement} node The node where to trap the focus
 */
export function trap(node) {
  if (FOCUS_TRAP) {
    throw new Error('The focus is already trapped, you must release it first')
  }

  const originalFocus = document.activeElement
  const originalTabIndex = node.tabIndex
  node.tabIndex = 0
  node.focus()

  const focusin = (e) => (
    !node.contains(e.target)
    && (focusChild(node, FOCUS_DIRECTION) || node.focus())
  )

  document.addEventListener('focus', focusin, true)

  FOCUS_TRAP = {
    release(restoreOriginalFocus) {
      node.tabIndex = originalTabIndex
      document.removeEventListener('focus', focusin, true)

      if (restoreOriginalFocus) {
        originalFocus.focus()
      }

      FOCUS_TRAP = null
    }
  }
}
