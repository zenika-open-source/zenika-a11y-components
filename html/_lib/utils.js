export * as dom from './utils/dom.js'
export * from './utils/log.js'
export * from './utils/timming.js'
export * from './utils/visibility.js'
