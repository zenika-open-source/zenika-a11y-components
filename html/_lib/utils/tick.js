/**
 * Wait for a full run of the event loop
 *
 * Some browser, like Safari, can require a full run of the event loop to
 * update (reflow or repaint) the page, which can be necessary to perform
 * some effects like transitionning a CSS value after a change of an element
 * display property.
 * @returns {Promise}
 */
export const tick = () => new Promise(requestAnimationFrame)

/**
 * Wait for roughly a given amount of milliseconds
 *
 * @param {number} ms The number of milliseconds to wait
 * @returns {Promise}
 */
export const wait = (ms) => new Promise((done) => setTimeout(done, ms))
