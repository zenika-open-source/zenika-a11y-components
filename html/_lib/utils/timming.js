/**
 * A simple Promise that wait for document readyness.
 *
 * Because JS module are defered, we need to be carefule and wait for
 * `DOMContentLoaded` to be fired before accessing the DOM.
 */
export const ready = new Promise((resolve) => {
  function isReady() {
    window.removeEventListener('DOMContentLoaded', isReady)
    resolve()
  }

  window.addEventListener('DOMContentLoaded', isReady)

  if (
    document.readyState === 'interactive'
    || document.readyState === 'complete'
  ) {
    isReady()
  }
})

/**
 * Wait for a full run of the event loop
 *
 * Some browser, like Safari, can require a full run of the event loop to
 * update (reflow or repaint) the page, which can be necessary to perform
 * some effects like transitionning a CSS value after a change of an element
 * display property.
 * @returns {Promise}
 */
export const tick = () => new Promise(requestAnimationFrame)

/**
 * Wait for roughly a given amount of milliseconds
 *
 * @param {number} ms The number of milliseconds to wait
 * @returns {Promise}
 */
export const wait = (ms) => new Promise((done) => setTimeout(done, ms))
