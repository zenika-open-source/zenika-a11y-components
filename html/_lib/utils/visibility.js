import css from '../css.js'
import { tick, wait } from './timming.js'

const MAX_TRANSITION_DURATION = 1000 // 1s
const MIN_TRANSITION_DURATION = 10 // 0.01s

// UTILS ------------------------------------------------------------------------------------------
function awaitTransition(node, timeout = 1000) {
  return Promise.race([
    wait(timeout),
    new Promise((done) => (
      node.addEventListener('transitionend', done, { once: true })
    ))
  ])
}

function isIntersecting(node, target = null) {
  return new Promise((done) => {
    if (
      node === document.documentElement // We consider the root element to be always intersecting
      || node === document.body // We consider the body to be always intersecting
    ) {
      done(true)
      return
    }

    const observer = new IntersectionObserver((entries) => {
      observer.unobserve(node)
      done(entries[0].intersectionRatio > 0.8) // Must "see" 80% of the element to be true
    }, { root: target })
    observer.observe(node)
  })
}

// MAIN API ---------------------------------------------------------------------------------------

/**
 * @typedef {object} ShouldBeVisibleReason
 * @property {boolean} isInDocumentBody
 *   An element can only be visible if it is part of the document body.
 *   An element inside a document fragment can't be visible.
 * @property {boolean} isDisplayed
 *   An element can be visible only if its CSS `display` property is different than `none`
 * @property {boolean} isVisible
 *   An element can be visible only if its CSS `visibility` property is different than `hidden`
 * @property {boolean} isOpaque
 *   An element can be visible only if its CSS `opacity` property is greater than `0`
 * @property {boolean} isWideEnough
 *   An element can be visible only if computed bounding box width is greater than `0`
 * @property {boolean} isHighEnough
 *   An element can be visible only if computed bounding box height is greater than `0`
 * @property {boolean} isInViewport
 *   An element can only be visible if it's inside the viewport (at least 80% of it)
 * @property {boolean} isInParentVisibleScrollingArea
 *   An element can only be visible if it's in the visible part of
 *   the scrolling area of all its parents (at least 80% of it)
 * @property {boolean} isInsideDisplayedParent
 *   An element can be visible only if all its parents are themselves displayed
 * @property {boolean} isInsideVisibleParent
 *   An element can be visible only if all its parents are themselves visible
 * @property {boolean} isInsideOpaqueParent
 *   An element can be visible only if all its parents are themselves opaque
 * @property {boolean} isInsideWideEnoughParent
 *   An element can be visible only if all its parents are themselves wide enough
 * @property {boolean} isInsideHighEnoughParent
 *   An element can be visible only if all its parents are themselves high enough
 */

/**
 * Check if the provided node should be visible
 *
 * Note this is only a hint, not an absolute certainty. Even if we check for
 * the obvious reason a DOM node should be visible, some weird CSS properties
 * combination can fully hide the element.
 *
 * @param {Element} node The DOM node we want to check for audibility
 * @returns {Promise<[boolean, ShouldBeVisibleReason]>}
 */
export async function shouldBeVisible(node) {
  const isInDocumentBody = document.body.contains(node)

  const style = window.getComputedStyle(node)
  const rect  = node.getBoundingClientRect()
  const isDisplayed  = style.display !== 'none'
  const isVisible    = style.visibility !== 'hidden'
  const isOpaque     = style.opacity > 0
  const isWideEnough = rect.width > 0
  const isHighEnough = rect.height > 0
  const isInViewport = await isIntersecting(node)

  let parent = node.parentNode
  let isInParentVisibleScrollingArea = await isIntersecting(node, parent)

  let isInsideDisplayedParent  = true
  let isInsideVisibleParent    = true
  let isInsideOpaqueParent     = true
  let isInsideWideEnoughParent = true
  let isInsideHighEnoughParent = true

  while (parent && parent instanceof Element) {
    const parentStyle = window.getComputedStyle(node)
    const parentRect  = node.getBoundingClientRect()

    isInsideDisplayedParent  = isInsideDisplayedParent  && parentStyle.display !== 'none'
    isInsideVisibleParent    = isInsideVisibleParent    && parentStyle.visibility !== 'hidden'
    isInsideOpaqueParent     = isInsideOpaqueParent     && parentStyle.opacity > 0
    isInsideWideEnoughParent = isInsideWideEnoughParent && parentRect.width > 0
    isInsideHighEnoughParent = isInsideHighEnoughParent && parentRect.height > 0
    isInParentVisibleScrollingArea
      =  isInParentVisibleScrollingArea
      && await isIntersecting(parent, parent.parentNode)

    parent = parent.parentNode

    // It can be expensive to walk up the parent tree so
    // we release the JS event loop on a regular basis
    await tick()
  }

  return [
    (
      isInDocumentBody
      && isDisplayed
      && isVisible
      && isOpaque
      && isWideEnough
      && isHighEnough
      && isInViewport
      && isInsideDisplayedParent
      && isInsideVisibleParent
      && isInsideOpaqueParent
      && isInsideWideEnoughParent
      && isInsideHighEnoughParent
      && isInParentVisibleScrollingArea
    ), {
      isInDocumentBody,
      isDisplayed,
      isVisible,
      isOpaque,
      isWideEnough,
      isHighEnough,
      isInViewport,
      isInsideDisplayedParent,
      isInsideVisibleParent,
      isInsideOpaqueParent,
      isInsideWideEnoughParent,
      isInsideHighEnoughParent,
      isInParentVisibleScrollingArea
    }
  ]
}

/**
 * @typedef {object} ShouldBeAudibleReason
 * @property {boolean} isInDocumentBody
 *   An element can only be audible if it is part of the document body.
 *   An element inside a document fragment shouldn't be audible.
 * @property {boolean} isNotAriaHidden
 *   An element with the `aria-hidden` attribute is **NOT** audible,
 *   so it shouldn't have that attribute to be audible.
 * @property {boolean} isNotHidden
 *   An element with the `hidden` attribute should NOT be audible,
 *   so it shouldn't have that attribute to be audible.
 * @property {boolean} isPerceivable
 *   An element is perceivable from an audio standpoint if it isn't
 *   hard hidden with CSS, which mean that at minimum its `display`
 *   style is not `none` and its `visibility` style is not `hidden`.
 * @property {boolean} isNotInsideAriaHiddenParent
 *   An element inside a parent with the `aria-hidden` attribute
 *   is NOT audible, so **NO** parent node up to the top of the DOM tree
 *   should have the `aria-hidden` attribute
 * @property {boolean} isNotInsideHiddenParent
 *   An element inside a parent with the `hidden` attribute should NOT
 *   be audible, so **NO** parent node up to the top of the DOM tree
 *   should have the `hidden` attribute
 * @property {boolean} isInsidePerceivableParent
 *   An element is perceivable from an audio standpoint if it isn't
 *   inside a parent that is hard hidden with CSS, which mean that at
 *   minimum all parent up to the top of the DOM tree have a `display`
 *   style that is not `none` and a `visibility` style that is not `hidden`.
 */

/**
 * Check if the provided node should be audible
 *
 * Note this is only a hint, not an absolute certainty. Even if we check for the
 * obvious reason a DOM node should not be audible, each browser/screenreader
 * combination as unique behaviors that can be hard to spot and we can end up in
 * cases that we could think a node is audible but it isn't in practice or vice versa.
 *
 * @param {Element} node The DOM node we want to check for audibility
 * @returns {Promise<[boolean, ShouldBeAudibleReason]>}
 */
export async function shouldBeAudible(node) {
  const isInDocumentBody = document.body.contains(node)
  const isNotAriaHidden = !node.getAttribute('aria-hidden')
  const isNotHidden = !node.getAttribute('hidden')

  const style = window.getComputedStyle(node)
  const isPerceivable
    =  style.display    !== 'none'
    && style.visibility !== 'hidden'

  let isNotInsideAriaHiddenParent = true
  let isNotInsideHiddenParent = true
  let isInsidePerceivableParent = true

  let parent = node.parentNode

  while (parent && parent instanceof Element) {
    const parentStyle = window.getComputedStyle(parent)
    isInsidePerceivableParent
      =  isInsidePerceivableParent
      && parentStyle.display !== 'none'
      && parentStyle.visibility !== 'hidden'

    isNotInsideAriaHiddenParent
      =  isNotInsideAriaHiddenParent
      && !parent.matches('[aria-hidden]')

    isNotInsideHiddenParent
      =  isNotInsideHiddenParent
      && !parent.matches('[hidden]')

    parent = parent.parentNode

    // It can be expensive to walk up the parent tree so
    // we release the JS event loop on a regular basis
    await tick()
  }

  return [
    (
      isInDocumentBody
      && isNotAriaHidden
      && isNotHidden
      && isPerceivable
      && isNotInsideAriaHiddenParent
      && isNotInsideHiddenParent
      && isInsidePerceivableParent
    ), {
      isInDocumentBody,
      isNotAriaHidden,
      isNotHidden,
      isPerceivable,
      isNotInsideAriaHiddenParent,
      isNotInsideHiddenParent,
      isInsidePerceivableParent
    }
  ]
}

/**
 * @typedef {object} ShowOptions
 * @property {string} [softHideClassName]
 *   Define a custom name for the CSS class representing the soft hide state.
 *   By default we use the css.HIDE class, but it can be useful to walk the
 *   whole show process to deal with different custom soft hide states.
 * @property {string} [className]
 *   A class name that represent a specific visible state. This class name
 *   is set just before the `softHideClassName` to be removed from the element,
 *   this allow to ensure that CSS transition are define and run ASAP.
 * @property {boolean|number} [transition]
 *   Indicate if we must wait for a transition (true) or how long we must
 *   wait before resolving the show process. When set to `true` we are
 *   waiting up to 1s for a `transitionend` event on the node before resolving
 *   the show process. When set to a number greater than 10, we wait for the first
 *   to happen: a `transitionend` event on the element or the number of milliseconds
 *   provided. By default, we do not wait for any transition and the show process
 *   is resolved as soon as attributes and classNames are set properly.
 */

/**
 * show a DOM element
 *
 * @param {Element} node
 * @param {ShowOptions} options
 * @returns {Promise}
 */
export async function show(node, options = {}) {
  const softHideClassName = options.softHideClassName || css.HIDE

  // hard hide, we need to transition to soft hide first
  if (node.hidden) {
    node.classList.add(softHideClassName) // Just to be sure its here
    node.hidden = false
    node.removeAttribute('aria-hidden')
    await tick()
  }

  // soft hide, let's make it visible
  if (node.classList.contains(softHideClassName)) {
    // Custom "visible" class name
    if (options.className) {
      node.classList.add(options.className)
    }

    node.classList.remove(softHideClassName)

    if (options.transition) {
      await awaitTransition(
        node,
        options.transition > 1 ? options.transition : MAX_TRANSITION_DURATION
      )
    }
  }
}

/**
 * Synchronously set DOM state reflecting a show final state
 *
 * @param {Element} node
 * @param {ShowOptions} options
 */
show.setImmediateShowState = (node, options = {}) => {
  node.hidden = false
  node.removeAttribute('aria-hidden')
  node.classList.remove(options.softHideClassName || css.HIDE)

  if (options.className) {
    node.classList.add(options.className)
  }
}

/**
 * @typedef {object} HideOptions
 * @property {boolean} [soft]
 *   When set to `true`, we stop the hide process in the soft hide state
 * @property {string} [softHideClassName]
 *   Define a custom name for the CSS class representing the soft hide state.
 *   By default we use the css.HIDE class, but it can be useful to walk the
 *   whole hide process to deal with different custom soft hide states.
 * @property {boolean|number} [transition]
 *   Indicate if we must wait for a transition (true) or how long we must
 *   wait before transitioning the hide process from soft to hard. When
 *   set to `true` we are waiting up to 1s for a `transtionend` event on
 *   the node before moving to hard hide. When set to a number greater than 10,
 *   we wait for the first to happen: a transitionend event on the element
 *   or the number of milliseconds provided. By default, we do not wait for
 *   any transition and the hide process move from soft to hide immediately.
 *   NOTE: If soft is set to true and transition is defined, the hide process
 *   will resolved once the transition is completed.
 */

/** hide a DOM element
 * @param {Element} node
 * @param {HideOptions} options
 * @returns {Promise}
 */
export async function hide(node, options = {}) {
  node.classList.add(options.softHideClassName || css.HIDE)
  await tick()

  if (options.transition) {
    const timeout = Math.max(Number(options.transition) || 0, 0)
    await awaitTransition(
      node,
      timeout > MIN_TRANSITION_DURATION ? timeout : MAX_TRANSITION_DURATION
    )
  }

  if (options.soft) {
    return
  }

  node.setAttribute('aria-hidden', true)
  node.hidden = true
  await tick() // Browser reflow is forced before resolving the process
}

/**
 * Synchronously set DOM state reflecting a hide final state
 *
 * @param {Element} node
 * @param {HideOptions} options
 */
hide.setImmediateHideState = (node, options = {}) => {
  if (!options.soft) {
    node.hidden = true
    node.setAttribute('aria-hidden', true)
  }

  node.classList.add(options.softHideClassName || css.HIDE)
}
