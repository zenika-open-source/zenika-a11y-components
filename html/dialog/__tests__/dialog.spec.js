import { jest } from '@jest/globals' // https://jestjs.io/docs/ecmascript-modules
import jestAxe from 'jest-axe'
import userEvent from '@testing-library/user-event'
import { tick } from '../../_lib/utils'
import Dialog from '../dialog'

const event = userEvent.default
const { axe } = jestAxe
jest.spyOn(console, 'error').mockImplementation(() => {})

describe('Dialog', () => {
  let node

  beforeEach(() => {
    document.body.innerHTML = `
      <div aria-label="A nice label for testing">
        <strong>Will move</strong> into an overlay
      </div>`

    node = document.body.firstElementChild
  })

  test('new Dialog(node) // With NO aria-label', () => {
    node.removeAttribute('aria-label')
    const dialog = new Dialog(node)

    expect(dialog).toBeInstanceOf(Dialog)
    expect(dialog.dialog).toBe(node)
    expect(dialog.node).toMatchSnapshot()

    expect(console.error).toHaveBeenCalledWith(
      expect.stringContaining('Dialog box MUST have an accessible label'),
      expect.anything(),
      expect.anything(),
      expect.anything(),
      expect.anything()
    )
  })

  test('new Dialog(node[aria-label])', async () => {
    const dialog = new Dialog(node)

    expect(dialog).toBeInstanceOf(Dialog)
    expect(dialog.node).toMatchSnapshot()
    expect(await axe(node)).toHaveNoViolations()
    expect(await axe(dialog.node)).toHaveNoViolations()
  })

  test('Dialog::show()', async () => {
    const dialog = new Dialog(node)
    await dialog.show()

    expect(dialog.node).toMatchSnapshot()
    expect(await axe(node)).toHaveNoViolations()
    expect(await axe(dialog.node)).toHaveNoViolations()
  })

  test('Dialog::hide()', async () => {
    const dialog = new Dialog(node)
    await dialog.show()
    await dialog.hide()

    expect(dialog.node).toMatchSnapshot()
    expect(await axe(node)).toHaveNoViolations()
    expect(await axe(dialog.node)).toHaveNoViolations()
  })

  test('Dialog.close()', async () => {
    const dialog = new Dialog(node)
    await dialog.show()
    await Dialog.close()

    expect(dialog.node).toMatchSnapshot()
    expect(await axe(node)).toHaveNoViolations()
    expect(await axe(dialog.node)).toHaveNoViolations()
  })

  test('Dialog.getOpen()', async () => {
    expect(Dialog.getOpen()).toBe(null)

    const dialog = new Dialog(node)
    await dialog.show()

    expect(Dialog.getOpen()).toBe(dialog)

    await dialog.hide()

    expect(Dialog.getOpen()).toBe(null)
  })

  test('Keyboard ESC close dialog boxes', async () => {
    const dialog = new Dialog(node)
    await dialog.show()

    event.keyboard('{esc}')
    await tick()

    expect(dialog.node).toMatchSnapshot()
    expect(await axe(node)).toHaveNoViolations()
    expect(await axe(dialog.node)).toHaveNoViolations()
  })
})
