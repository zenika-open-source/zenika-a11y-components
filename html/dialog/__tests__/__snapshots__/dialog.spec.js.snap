// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog Dialog.close() 1`] = `
<div
  aria-hidden="true"
  class="dialog overlay hidden"
  hidden=""
  style="position: fixed; left: 0px; top: 0px; width: 1024px; height: 768px;"
>
  <div
    aria-label="A nice label for testing"
    aria-modal="true"
    role="dialog"
    tabindex="-1"
  >
    
        
    <strong>
      Will move
    </strong>
     into an overlay
      
  </div>
</div>
`;

exports[`Dialog Dialog::hide() 1`] = `
<div
  aria-hidden="true"
  class="dialog overlay hidden"
  hidden=""
  style="position: fixed; left: 0px; top: 0px; width: 1024px; height: 768px;"
>
  <div
    aria-label="A nice label for testing"
    aria-modal="true"
    role="dialog"
    tabindex="-1"
  >
    
        
    <strong>
      Will move
    </strong>
     into an overlay
      
  </div>
</div>
`;

exports[`Dialog Dialog::show() 1`] = `
<div
  class="dialog overlay"
  style="position: fixed; left: 0px; top: 0px; width: 1024px; height: 768px;"
>
  <div
    aria-label="A nice label for testing"
    aria-modal="true"
    role="dialog"
    tabindex="0"
  >
    
        
    <strong>
      Will move
    </strong>
     into an overlay
      
  </div>
</div>
`;

exports[`Dialog Keyboard ESC close dialog boxes 1`] = `
<div
  aria-hidden="true"
  class="dialog overlay hidden"
  hidden=""
  style="position: fixed; left: 0px; top: 0px; width: 1024px; height: 768px;"
>
  <div
    aria-label="A nice label for testing"
    aria-modal="true"
    role="dialog"
    tabindex="-1"
  >
    
        
    <strong>
      Will move
    </strong>
     into an overlay
      
  </div>
</div>
`;

exports[`Dialog new Dialog(node) // With NO aria-label 1`] = `
<div
  aria-hidden="true"
  class="dialog overlay hidden"
  hidden=""
>
  <div
    aria-modal="true"
    role="dialog"
  >
    
        
    <strong>
      Will move
    </strong>
     into an overlay
      
  </div>
</div>
`;

exports[`Dialog new Dialog(node[aria-label]) 1`] = `
<div
  aria-hidden="true"
  class="dialog overlay hidden"
  hidden=""
>
  <div
    aria-label="A nice label for testing"
    aria-modal="true"
    role="dialog"
  >
    
        
    <strong>
      Will move
    </strong>
     into an overlay
      
  </div>
</div>
`;
